src/Microsoft.Dnx.DesignTimeHost/Models/OutgoingMessages/DiagnosticMessageView.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.Dnx.Runtime;
+
+namespace Microsoft.Dnx.DesignTimeHost.Models.OutgoingMessages
+{
+    public class DiagnosticMessageView
```

Comment:
Yeah it's safer to do so. Now the `private class Comparer : IEqualityComparer<DiagnosticMessageView>` in `DiagnosticsListMessage` actually helps to compare.